### ISSUE TITLE
Should This Repo Be Public or Private

### ISSUE THREAD

**Issue by benhutton** [desiringgod/guides#52]
This is a public repo, forked from thoughtbot's guides repo.

We left it public because it was essentially meant to be an open source contribution, perhaps a hiring tool, and perhaps a source of interest to those wanting to know how we do things at DG.

The goal was to document style and best practices. How we use ruby, rails, javascript, css, html, etc. How we use tools like git, github, etc.

Since then, that vision has been diluted and our team has significantly expanded. With that comes an increased desire for documentation.

Right now, it seems that we are treating this repo as the place to document anything that isn't repo specific. That's great, but the catch-all nature of it is philosophically at odds with how we originally conceived of this as public.

Beyond that, I think that the public-ness of the repo at this point does more harm than good. We don't really gain that much, if at all, from it being public. The cost is multiple. Some people aren't aware (or keenly aware) of it being public, so there's the risk of confidential things being made public. For those that are aware, there's mental overhead of thinking about whether something should be made public.

Right now, the question we ask to decide "does something go in the guides repo" does not significantly include, "do we want this public for all the world to see." It is simply, "does this belong more appropriately to a particular app repo?"

Thus, I propose that we either own the publicness of this repo and add that explicitly into our documentation process (meaning: create a centralized location for non-repo-specific docs and issues) OR we make this repo private.

**Comment by benhutton** [comment]
Make this repo private

**Comment by benhutton** [comment]
Keep this repo public and create a new place for confidential non-repo-specific docs and issues

**Comment by benhutton** [comment]
Do nothing and maintain the status quo

**Comment by pjmelling** [comment]
I'm fine with a best practices guide being public. We're starting to put specific configuration details into the guides. @benhutton you did an excellent job (in my estimation) of describing the situation. I'm all for contributing to the community, but we need leadership in that area for it to be worthy.

**Comment by hchasey** [comment]
@benhutton – do you have some examples of "confidential non-repo-specific docs and issues"? 

@pjmelling – can you explain what you mean by "we need leadership in that area for it to be worthy"?

@isaachvazquez @pjmelling – you both voted for both "Make this repo private" and "Keep this repo public...."  

@benhutton @gaganawhad – since you two might know the history of this guide the best, on this note "The goal was to document style and best practices. How we use ruby, rails, javascript, css, html, etc. How we use tools like git, github, etc." I see that the [Workflow section](https://github.com/desiringgod/guides/tree/master/workflow) was left in the fork and the [Security](https://github.com/thoughtbot/guides/tree/master/security) and [Protocol](https://github.com/thoughtbot/guides/tree/master/protocol) sections were removed or not yet present. Can you expand at all on those decisions? 

PS My vote is still pending 😄

**Comment by isaachvazquez** [comment]
@hchasey I don't have any preferences either way, so I cancelled my vote by voting for both. :)

**Comment by benhutton** [comment]
@hchasey well, we've done a good job of keeping things that are critically confidential out of here. We don't put credentials or keys to anything in here.

https://github.com/desiringgod/guides/pull/48 might be the closest we've come to "confidential" in this repo, though I suppose that is arguable.

Here's another example: https://github.com/desiringgod/www/wiki/Cloud-Storage-&-Distribution#current-s3-buckets

Perhaps more significantly is that these guides and issues can't properly interact with code in our private repos. Pasting code into the guides exposes private information. Linking to code from the guides lessens their usefulness and makes them not very public-friendly. Same with slack conversations, trello cards, etc. The more we interlink with private documents, the less intelligible this becomes.

**Comment by gaganawhad** [comment]
@hchasey re: `Security` and `Protocol` decision, I am pretty sure they haven't made it to this repo _yet_ because the document on thoughtbot was too / mostly specific them. We mostly had the mindset of keeping the broader more applicable things around and then come to specific things later, if required.

**Comment by gaganawhad** [comment]
also, I agree with @pjmelling that I am good with our best-practices being public, and also sort of make a distinction on what a best practice is vs what a documentation is. Our documentation may be more specific to us, sensitive, and might not help the community much, but on the other hand keeping best practices open might not hurt.

**Comment by joeosburn-desiringgod** [comment]
Is there really enough benefit in keeping a public best practices repo? If we go with that approach, we'd have to maintain two separate docs repos, simply in order to keep one public. Now, when we want to add or look up documentation, we have to figure out which of the two repos to use. All this to keep one of the two repos public? Which has downsides in addition to theoretical/speculative upsides.

**Comment by ryanjames** [comment]
I agree with Joe. It seems like a hassle to try maintaining two separate guides and always needing to evaluate what goes where for the sake of sharing our best practices with the world.
